### DISCUSSION TITLE
It will be nice to have flag that allows to generate getters for all models (esp. Request one)

### DISCUSSION THREAD

**Discussion by vitalii-marchenko** [kobylynskyi/graphql-java-codegen#740] [Ideas]
Hi! Thank you for such a great project!
I just wanted to ask/suggest if it is possible to generate models in Java with getters? Currently, I can see that only setters are generated, and to access the fields you should work with Map<String, Object> input.
It is not very convenient when you are writing integration API tests and want to validate that the request has been processed correctly.
Probably I'm doing something incorrectly?

Thanks again for the great plugin!

**Comment by jxnu-liguobin** [comment]
@markony647 Yes , you're right. There is no `get` method generated for it.

**Comment by kobylynskyi** [comment]
@markony647 thanks for your message.
The thing is that input is stored as `Map<String, Object>` (due to the fact that at the moment of generation we do not know which fields the user will set).
That's why the only method we could add is:
```java
public Object get(String fieldName) {
    return input.get(fieldName);
}
```
This sounds not that helpful, right? Because probably you are already doing something like: 
```java
(FieldType) request.getInput().get("fieldName");
```
Please let me know your thoughts.

**Reply by vitalii-marchenko** [reply]
hi Bogdan! yes, currently I'm doing something like you mentioned above. 
One more possible workaround for tests is to create custom Java models with getters and then work with them. 
```` 
GeneratedSignupQueryRequest generatedSignupQueryRequest = resolveObject(GeneratedSignupQueryRequest.class, customSignupRequest);
```` 
in that case, I build customSignupRequest and send it to the server so then I can do any assertions like 
```` 
assertThat(myResponse.getFirstName()).isEqualTo(customSignupRequest.getFirstName())
```` 
But this approach requires to generate too much models that is also not super convenient.
